packages/mdc-tabs/tab-bar-scroller/foundation.ts

```diff
@@ -35,55 +39,62 @@ export default class MDCTabBarScrollerFoundation extends MDCFoundation {
-      getOffsetWidthForTabBar: () => /* number */ 0,
-      setTransformStyleForTabBar: (/* value: string */) => {},
-      getOffsetLeftForEventTarget: (/* target: EventTarget */) => /* number */ 0,
-      getOffsetWidthForEventTarget: (/* target: EventTarget */) => /* number */ 0,
+      addClass: () => undefined,
+      removeClass: () => undefined,
+      eventTargetHasClass: () => false,
+      addClassToForwardIndicator: () => undefined,
+      removeClassFromForwardIndicator: () => undefined,
+      addClassToBackIndicator: () => undefined,
+      removeClassFromBackIndicator: () => undefined,
+      isRTL: () => false,
+      registerBackIndicatorClickHandler: () => undefined,
+      deregisterBackIndicatorClickHandler: () => undefined,
+      registerForwardIndicatorClickHandler: () => undefined,
+      deregisterForwardIndicatorClickHandler: () => undefined,
+      registerCapturedInteractionHandler: () => undefined,
+      deregisterCapturedInteractionHandler: () => undefined,
+      registerWindowResizeHandler: () => undefined,
+      deregisterWindowResizeHandler: () => undefined,
+      getNumberOfTabs: () => 0,
+      getComputedWidthForTabAtIndex: () => 0,
+      getComputedLeftForTabAtIndex: () => 0,
+      getOffsetWidthForScrollFrame: () => 0,
+      getScrollLeftForScrollFrame: () => 0,
+      setScrollLeftForScrollFrame: () => undefined,
+      getOffsetWidthForTabBar: () => 0,
+      setTransformStyleForTabBar: () => undefined,
+      getOffsetLeftForEventTarget: () => 0,
+      getOffsetWidthForEventTarget: () => 0,
     };
+    // tslint:enable:object-literal-sort-keys
   }
 
-  constructor(adapter) {
-    super(Object.assign(MDCTabBarScrollerFoundation.defaultAdapter, adapter));
+  private pointerDownRecognized_ = false;
+  private currentTranslateOffset_ = 0;
+  private focusedTarget_: HTMLElement | null = null;
+  private layoutFrame_ = 0;
+  private scrollFrameScrollLeft_ = 0;
 
-    this.pointerDownRecognized_ = false;
-    this.currentTranslateOffset_ = 0;
-    this.focusedTarget_ = null;
-    this.layoutFrame_ = 0;
-    this.scrollFrameScrollLeft_ = 0;
-    this.forwardIndicatorClickHandler_ = (evt) => this.scrollForward(evt);
-    this.backIndicatorClickHandler_ = (evt) => this.scrollBack(evt);
+  private readonly forwardIndicatorClickHandler_: SpecificEventListener<'click'>;
+  private readonly backIndicatorClickHandler_: SpecificEventListener<'click'>;
+  private readonly resizeHandler_: SpecificEventListener<'resize'>;
+  private readonly interactionHandler_: SpecificEventListener<InteractionEventType>;
+
+  constructor(adapter?: Partial<MDCTabBarScrollerAdapter>) {
+    super({...MDCTabBarScrollerFoundation.defaultAdapter, ...adapter});
+
+    this.forwardIndicatorClickHandler_ = (evt: MouseEvent) => this.scrollForward(evt);
+    this.backIndicatorClickHandler_ = (evt: MouseEvent) => this.scrollBack(evt);
     this.resizeHandler_ = () => this.layout();
     this.interactionHandler_ = (evt) => {
```

Comment:
I get a TS error for `evt` being implicitly any. I think you need to change it to `(evt: SpecificEventListener<InteractionEventType>) => `

Reply:
Curious, since we seem to get away with something similar to this in text field and select's icon foundations.

Reply:
Weird, I'm not seeing that error. In fact, I was able to remove `: MouseEvent` a few lines up.

Did you re-run `git pull && npm i`?

Reply:
Running on Linux:
```
npm cache clear -f
npm i
npm run build rn # causes error

node -v 10.13.0
npm -v 6.5.0
```

Reply:
Talked offline...seems like it is just my machine. @acdvorak tried on both his mac and linux machines. Ignoring this issue for now, as the issue is local to my machine.

Reply:
After discussing this over chat, it sounds like the issue is only occurring on one of Matt's machines. The build passes on every other machine, including Travis CI. So it's probably fine.